### PR TITLE
Add tests for LevelExtractor and ResampleFFT

### PR DESCRIPTION
**Pull request by seeker-Liu**

### CODE REVIEW

File: test/src/unittests/synthesis/test_resamplefft_standard.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        # Due to the nature of FFT resample, it is improper to test with random input.
+        import os.path as path
+        import pathlib
+        with np.load(path.join(pathlib.Path(__file__).resolve().parent, "dubstep_first_second.npz")) as data:
+            raw = data["raw"]
+            expected = data["expected"]
+            r = ResampleFFT()
+            r.configure(inSize=44100, outSize=32768)
+            output = r(raw)
+            self.assertAlmostEqualVectorFixedPrecision(output, expected, 3)  # The error is a little bit too big
```
Comment (dbogdanov) [inline test/src/unittests/synthesis/test_resamplefft_standard.py:59]: Agreed, would be nice to find out the reason.

File: test/src/unittests/synthesis/test_resamplefft_standard.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+    def test_odd_length(self):
+        import os.path as path
+        import pathlib
+        with np.load(path.join(pathlib.Path(__file__).resolve().parent, "mozart_c_major_fragment.npz")) as data:
```
Comment (dbogdanov) [inline test/src/unittests/synthesis/test_resamplefft_standard.py:65]: Same here for the location of the data file.

**Review by dbogdanov — CHANGES_REQUESTED** [review]

File: test/src/unittests/synthesis/test_resamplefft_standard.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    def test_empty(self):
+        empty = np.array([], dtype=np.single)
+        r = ResampleFFT()
+        with self.assertRaises(RuntimeError):
+            r(empty)
+
+    def test_silence(self):
+        silence = np.zeros(44100, dtype=np.single)
+        r = ResampleFFT()
+        r.configure(outSize=10492 * 3)
```
[the remark this answers is not in the document]
Comment (seeker-Liu) [reply test/src/unittests/synthesis/test_resamplefft_standard.py:34]: It is actually the arbitrary value I used like 42. No rational behind

File: test/src/unittests/extractor/test_levelextractor.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        res = LevelExtractor()(silence_vec)
+        self.assertEqual(len(res), 1)
+        self.assertAlmostEqualFixedPrecision(res[0], 0, 5)
+
+    def test_illegal_param(self):
+        def param_setting_entries(frameSize, hopSize):
+            def param_setting_lambda(frameSize, hopSize):
+                le = LevelExtractor()
+                le.configure(frameSize=frameSize, hopSize=hopSize)
+            return functools.partial(param_setting_lambda, frameSize, hopSize)
```
[the remark this answers is not in the document]
Comment (seeker-Liu) [reply test/src/unittests/extractor/test_levelextractor.py:42]: I am not aware of the existence of this assertion.

**Pull request merged**